extensions/vscode/src/views/logs.ts

```diff
@@ -229,6 +233,17 @@ export class LogsTreeDataProvider implements TreeDataProvider<LogsTreeItem> {
           this.refresh();
         },
       );
+
+      // Register for some specific events we need to handle differently
+      this._stream.register(
+        "publish/restoreEnv/status",
+        (msg: EventStreamMessage) => {
+          const stage = this.stages.get("publish/restoreEnv");
+          if (stage && isPublishRestoreEnvStatus(msg)) {
+            stage.activeLabel = `${RestoringEnvironmentLabel} - ${restoreMsgToStatusSuffix(msg)}`;
```

Comment:
I'm having a bit of trouble understanding what is happening here.

Reply:
We're updating the label for the Restore Environment stage when we receive a status event for it? To do this, we are updating the child tree item for the stage, which then causes a refresh, which repaints the change. Perhaps we should zoom real quick...